### PR TITLE
Add the first draft of heat-flow-L0 module.

### PR DESCRIPTION
**Pull request by alarshi**
This PR breaks the existing heat-flow-module-L0 by @mgouiza into several smaller notebooks. Additionally, two concepts---cooling of oceanic lithosphere and plate cooling model---are added in this module.
I tested that the notebooks render correctly on my laptop and most of them also work on the HubZero platform. However, due to our current limitation to download the new python modules, I was not able to check three of the notebooks in this PR.
Nevertheless, I would say that they are ready for your review! :)

### CODE REVIEW

**Review by ljhwang — COMMENTED** [review]
Thanks for the initial upload. Looks good.

File: bin/heat-flow-module-L0/8_cooling_of_oceanic_lithosphere.ipynb [hunk 1]
```diff
@@ -0,0 +1,398 @@
+{
```
Comment (naliboff) [inline bin/heat-flow-module-L0/8_cooling_of_oceanic_lithosphere.ipynb:1]: <p>Rerun notebook to fix this error?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/5/discussion/#comment-1805445113'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (naliboff) [inline bin/heat-flow-module-L0/8_cooling_of_oceanic_lithosphere.ipynb:1]: <p>Related to error in the cell above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/5/discussion/#comment-1805445114'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

File: bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb [hunk 2]
```diff
@@ -0,0 +1,257 @@
+{
```
Comment (naliboff) [inline bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    def fourier_term (n, t):
</pre><p>A general comment - do we want to follow python best practices here and add proper documentation for each function?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/5/discussion/#comment-1805445116'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (naliboff) [inline bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Now, let's plot the heat flow due to the plate model with max ltuhisphere
</pre><p>lithosphere is misspelled at the end of the sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/5/discussion/#comment-1805445118'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

**Comment by naliboff** [comment]
@alarshi - I added a few comments on the two new modules (half-space and plate cooling) when you have a chance to address them. On my last to do a check on all the other files, although I imagine they should be in good shape!

File: bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb (hunk 2 above)
Comment (alarshi) [reply bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb:1]: Good point!I will add the documentation of all functions.

Comment (alarshi) [reply bin/heat-flow-module-L0/9_plate_cooling_model_of_lithosphere.ipynb:1]: Oh no, this was wrong on so many levels :D Thanks for catching this!

File: bin/heat-flow-module-L0/3_thermal_conductivity_from_geotherms.ipynb [hunk 3]
```diff
@@ -0,0 +1,432 @@
+{
```
Comment (naliboff) [inline bin/heat-flow-module-L0/3_thermal_conductivity_from_geotherms.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.    # Create a function to convert depth to pressure
</pre><p>Add function documentation here? I realize this is just a single line, but perhaps best to be consisent.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/5/discussion/#comment-1806893383'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

**Comment by naliboff** [comment]
@alarshi - Thanks for the revisions! I saw one more place where function documentation (for two functions) could be added, but these were places where the function was just a single line so easy to miss. Once that is added, this is good to merge. Thanks again and all of this looks fantastic!

File: bin/heat-flow-module-L0/3_thermal_conductivity_from_geotherms.ipynb (hunk 3 above)
Comment (alarshi) [reply bin/heat-flow-module-L0/3_thermal_conductivity_from_geotherms.ipynb:1]: Aa oo, forgot to add those! Thank you for catching them. :)

**Comment by alarshi** [comment]
@naliboff : Thank you for catching that! I added the definitions in the last commit.

**Review by naliboff — APPROVED** [review]

**Pull request merged**